### PR TITLE
fix: got working again in new GitHub web UI

### LINKED ISSUE #161: 🐛 Bug: No longer works in latest GitHub web update
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/refined-saved-replies/issues?q=is%3Aissue+label%3A%22type%3A+bug%22) and found none that matched my issue.

### Expected

This should still work in the latest github.com UI refresh.

### Actual

The new UI is different enough from the old one that this extension no longer does anything 😬 

Screenshot of the new GitHub modal saved replies

### Additional Info

...annoyed that GitHub downgraded the experience to a _modal_ but you still can't have org or repo saved replies :angry:

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #161
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/refined-saved-replies/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/refined-saved-replies/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Gets it working in the new modal world with minimal changes.

### CODE REVIEW

File: src/content-script.ts [hunk 1]
```diff
@@ -74,17 +75,39 @@ async function main() {
...
+		throw new Error(`Element was never found 😔`);
+	}
+
+	const onOpenSavedRepliesButtonClick = async () => {
 		// 7. Add the new replies to the saved reply dropdown
-		const replyCategoriesDetailsMenus = document.querySelectorAll(
-			`markdown-toolbar details-menu[src^="/settings/replies?context="]`,
+		const replyCategoriesDetailsMenus = await getSoon(() =>
+			document.querySelectorAll(
+				`div.Overlay-body fuzzy-list focus-group div[data-view-component]`,
```
Comment (keithamus) [inline src/content-script.ts:83]: A more stable selector here would be `.Overlay-body .js-saved-reply-menu`

File: src/content-script.ts [hunk 2]
```diff
@@ -124,38 +165,83 @@ async function main() {
...
-					role: "menuitem",
-					target: "_blank",
-				}),
-			);
+			// 8. Add a second button at the bottom of the modal for adding more
+			// TODO: thanks for the heads up @keithamus :)
+			// https://github.com/primer/view_components/pull/2364
+			for (const modal of Array.from(
+				document.querySelectorAll<HTMLElement>(
+					"modal-dialog#saved_replies_menu_new_comment_field-dialog",
```
Comment (keithamus) [inline src/content-script.ts:170]: You could keep this stable with `:where(modal-dialog, dialog).js-saved-reply-container` which probably won't change.

File: src/content-script.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/content-script.ts:83]: Dangit, merged and _then_ saw these comments. Thanks! Filing a followup issue...

**Comment by JoshuaKGoldberg** [comment]
@allcontributors add @keithamus for review

File: src/content-script.ts [hunk 3]
```diff
@@ -124,38 +149,83 @@ async function main() {
...
+				),
+			)) {
+				// Also, because the modal is by default too tiny, let's make it bigger
+				modal.style.height = "100%";
+				modal.style.maxHeight = "calc(100vh - 5rem)";
```
Comment (keithamus) [inline src/content-script.ts:175]: https://github.com/primer/view_components/blob/97f03393977c9c82141eb5255fa3fd619a95e935/app/components/primer/alpha/dialog.rb#L26-L31 these are the sizes defined in the design system. So maybe add the class `.Overlay--size-xlarge`

File: src/content-script.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/content-script.ts:83]: #168

**Pull request merged**